### PR TITLE
Use jOOQ-3.16-compatible way of getting XSD class path

### PR DESCRIPTION
**Pull request by perlun**
The approach taken here should be fine on older jOOQ versions as well. Extra care was taken to use reflection and try to degrade gracefully on jOOQ < 3.12 releases as well.

(Includes #205 as well, so if this gets approved we can drop that other PR.)

### CODE REVIEW

File: src/main/groovy/nu/studer/gradle/jooq/JooqGenerate.java [hunk 1]
```diff
@@ -254,11 +264,20 @@ private void writeConfiguration(Configuration config, File file) {
...
+
+            if (Arrays.stream(jooqConstants.getDeclaredFields())
+                    .map(Field::getName)
+                    .anyMatch(s -> s.equals("CP_CODEGEN"))) {
+                // jOOQ >= 3.12.0; we can use the "new" approach of constructing the XSD class path.
+                return (String) jooqConstants.getDeclaredField("CP_CODEGEN").get(null);
+            } else {
+                // For older versions, we resort to the previous hardwired approach.
+                return "/xsd/" + jooqConstants.getDeclaredField("XSD_CODEGEN").get(null);
+            }
```
Comment (perlun) [inline src/main/groovy/nu/studer/gradle/jooq/JooqGenerate.java:280]: Tested with `3.15.1` (i.e. the version referenced by `build.gradle`) with great success. :tada: 

Note that the fix in this PR is not enough for full jOOQ 3.16 support; there is still the annoying `name()` issue (similar to, or perhaps even identical to, the one I reported in #204) that causes issues with jOOQ 3.16 even after this change is in place. I'll keep digging into that as well, but hope to submit a fix to that in a separate PR.

**Comment by etiennestuder** [comment]
Thanks.

File: src/main/groovy/nu/studer/gradle/jooq/JooqGenerate.java (hunk 1 above)
Comment (perlun) [reply src/main/groovy/nu/studer/gradle/jooq/JooqGenerate.java:281]: (#208 fixes the `name()` issue btw)

**Comment by etiennestuder** [comment]
This is fixed in the recently relased v7.0 of the gradle-jooq-plugin. The problems was due to a backward-incompatible change in jOOQ between 3.15 and 3.16.

**Pull request merged**